dbt/include/snowflake/macros/materializations/dynamic_table/materialization.sql

```diff
@@ -46,39 +45,39 @@
 {% macro dynamic_table_teardown(backup_relation, intermediate_relation, post_hooks) %}
 
     -- drop the temp relations if they exist to leave the database clean for the next run
-    {{ snowflake__get_drop_dynamic_table_sql(backup_relation) }}
-    {{ snowflake__get_drop_dynamic_table_sql(intermediate_relation) }}
+    {{ drop_relation_if_exists(backup_relation) }}
+    {{ drop_relation_if_exists(intermediate_relation) }}
 
     {{ run_hooks(post_hooks) }}
 
 {% endmacro %}
 
 
-{% macro dynamic_table_get_build_sql(existing_relation, target_relation, backup_relation, intermediate_relation) %}
+{% macro dynamic_table_build_sql(dynamic_table, existing_relation, backup_relation, intermediate_relation) %}
 
     {% set full_refresh_mode = should_full_refresh() %}
 
     -- determine the scenario we're in: create, full_refresh, alter, refresh data
     {% if existing_relation is none %}
-        {% set build_sql = snowflake__get_create_dynamic_table_as_sql(target_relation, sql) %}
+        {% set build_sql = snowflake__create_dynamic_table_sql(dynamic_table) %}
     {% elif full_refresh_mode or not existing_relation.is_dynamic_table %}
-        {% set build_sql = snowflake__get_replace_dynamic_table_as_sql(target_relation, sql, existing_relation, backup_relation, intermediate_relation) %}
+        {% set build_sql = snowflake__replace_dynamic_table_sql(dynamic_table, existing_relation, backup_relation, intermediate_relation) %}
     {% else %}
 
         -- get config options
```

Comment:
Dangling/incomplete comment?

Reply:
This is _very_ similar to the new built-in materialized view materialization in `dbt-core` (if you find/replace "dynamic table" with "materialized view", you might even say they're the same). That being said, I had these comments at the time to try and lead myself (and the next unfortunate soul) through this giant, nested, jinja-dwelling if block. But it goes back to your comment of "comments should say why, not what". And I think what this is suggesting is that maybe it should be its own macro, which would self-document what I'm doing here. I'll try some combination of all of that to make this better.

Reply:
That'd be rad!